project/src/components/main/main.tsx

```diff
@@ -1,52 +1,44 @@
-import { Offer } from '../app/app';
-import { Link } from 'react-router-dom';
 import { OffersList } from '../offers-list/offers-list';
 import { Map } from '../map/map';
-import { useAppSelector } from '../../hooks';
+import { useAppDispatch, useAppSelector } from '../../hooks';
 import { CitiesList } from '../cities-list/cities-list';
 import { OffersSort } from '../offers-sort/offers-sort';
 import { getCurrentOffers } from './main.utils';
+import { Header } from '../header/header';
+import { fetchOffersAction } from '../../store/api-actions';
+import { useEffect, useState } from 'react';
+import { AuthorizationStatus } from '../../types/authorization.types';
+import { Spinner } from '../spinner/spinner';
 
-interface MainProps {
-  offers: Offer[],
-}
+export function Main(): JSX.Element {
+  const dispatch = useAppDispatch();
+  const { authorizationStatus, offers } = useAppSelector((st) => st);
+  const [isDataLoaded, setIsDataLoaded] = useState(false);
+
+  const fetchData = async () => {
+    await dispatch(fetchOffersAction());
+    setIsDataLoaded(true);
+  };
+
+  useEffect(() => {
+    fetchData();
+  }, [isDataLoaded]);
```

Comment:
лишняя зависимость
```suggestion
  }, [fetchData]);
```
запрашиваем 2 раза данные - это плохо